semantics/src/main/scala/aqua/semantics/expr/func/ForSem.scala

```diff
@@ -75,18 +78,39 @@ class ForSem[S[_]](val expr: ForExpr[S]) extends AnyVal {
 
 object ForSem {
 
+  def itemOrPair[S[_]](nameOrPair: ForExpr.NameOrPair[S]): (String, Option[ForKeyValue]) =
+    nameOrPair match {
+      case Right(v) => (v.value, None)
+      case Left(k, v) => ("-iterable-", ForKeyValue(k.value, v.value).some)
```

Comment:
Why do this on semantics level? 
I propose to make `ForTag(item: Either[...], ...)` and let inliner deal with it.

Reply:
the problem is that I should generate `NextTag` with item name also

Reply:
Oh, I see. I think we could move generation of `next` to the inliner too. But not rn, okay.